### ISSUE TITLE
Rancher generates wrong docker command therefore taints are not applied on nodes.

### ISSUE THREAD

**Issue by vishal563** [rancher/rancher#36131]
**Rancher Server Setup**
- Rancher version: 2.6.3
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version:
- Cluster Type (Local/Downstream):
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:

   
**Describe the bug**

**To Reproduce**
 

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by vishal563** [comment]
Did anyone faced similar issue?

**Comment by KingJ** [comment]
I have also run in to this on Rancher 2.6.4, with the Registration UI for RKE nodes on a custom cluster outputting `--taint`. However, I can confirm that this doesn't result in the node being tainted on join and `--taints` must be used instead. `--taints` is also [the flag listed in the documentation](https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/rke-clusters/custom-nodes/agent-options/), so this seems to be a UI bug?

![image](https://user-images.githubusercontent.com/6966796/162918534-e69b79cd-0be7-47ef-988a-3ba2688bc537.png)

I've also opened up a support case referencing this ticket - 00343946.

**Comment by KingJ** [comment]
Bumping to prevent closure, as this is still an issue. I've been told the internal bug ticket for this is SURE-4481.

**Comment by KingJ** [comment]
Bumping to prevent closure, as this is still an issue.

**Comment by samjustus** [comment]
@KingJ  @vishal563 is the downstream a windows cluster?

**Comment by KingJ** [comment]
> @KingJ @vishal563 is the downstream a windows cluster?

In my case, no - Linux-based. 

I opened up a support case with Rancher a few months back, and they've acknowledged this bug. Not seen a fix released yet.

**Comment by HarrisonWAffel** [comment]
I would agree that this seems to be a UI bug. From what I can tell the registration command is generated [in this file](https://github.com/rancher/dashboard/blob/811c787a408e73aeaeacc4eb83ab521837ef140a/shell/edit/provisioning.cattle.io.cluster/CustomCommand.vue), and the taints are added to the command [here](https://github.com/rancher/dashboard/blob/811c787a408e73aeaeacc4eb83ab521837ef140a/shell/edit/provisioning.cattle.io.cluster/CustomCommand.vue#L68-L76). In that file that the flag `--taint` is being pushed to the UI, as opposed to the proper `--taints` flag. 

Curiously, this issue does not reproduce when registering custom RKE1 clusters which utilize windows worker nodes. In this case, the registration command will include the proper `--taints` flag both when generating the linux registration command as well as when generating the windows registration command.

**Comment by catherineluse** [comment]
I just tried to reproduce this issue in three ways, but in all three ways it seemed to work in the latest version of v2.6-head.

## 1. Node Template
I created an RKE1 node template with taints:
<img width="1348" alt="Screen Shot 2022-09-13 at 2 28 03 AM" src="https://user-images.githubusercontent.com/20599230/189874144-9722eb21-0526-478d-888e-02fa53dae756.png">
That resulted in the taint appearing correctly in the cluster:

<img width="1501" alt="Screen Shot 2022-09-13 at 2 59 21 AM" src="https://user-images.githubusercontent.com/20599230/189877730-11562d4e-6e5e-4b58-addb-3d9cefcb9490.png">




## 2. Node Pools
In the EC2 cluster's node pool section, I added a taint during cluster creation:

<img width="1248" alt="Screen Shot 2022-09-13 at 2 38 59 AM" src="https://user-images.githubusercontent.com/20599230/189877125-5d3c8113-d0aa-44c9-ada3-8e3c0bd4e441.png">
The taint showed up successfully:
<img width="1485" alt="Screen Shot 2022-09-13 at 2 58 55 AM" src="https://user-images.githubusercontent.com/20599230/189877611-95855714-c4f1-41d0-b7d6-bc2a9738a49f.png">

## 3. RKE1 Custom Cluster Creation
When I went to create a custom RKE1 cluster, the `--taints` was pluralized in the Docker command:
<img width="1254" alt="Screen Shot 2022-09-13 at 3 04 41 AM" src="https://user-images.githubusercontent.com/20599230/189877970-514600e1-9f6f-490c-b324-12bb11bc21d5.png">

I had a question about the steps to reproduce in the issue description. I don't see how one gets from the EC2 node pool cluster creation page to the docker command. When I create a node pool cluster on EC2, the form takes me straight to cluster management with no registration command shown. The registration command is only visible in the UI for custom clusters. So I'm wondering if the original issue meant to say it was a custom RKE1 cluster? And if not, how was the command obtained?

I believe the code that is being used for generating the docker commands for creating RKE1 clusters is being taken from an iFramed Ember page from the old UI. And the old UI seems to use the correct pluralization:

```
taints.forEach((taint) => {
      out += ` --taints ${ get(taint, 'key') }=${ get(taint, 'value') }:${ get(taint, 'effect') }`;
    });
```
At first I assumed this issue was about RKE1 due to the Docker command being used, but now it looks like the broken code is only in the dashboard, which isn't used for RKE1 provisioning. So I'd like to confirm the cluster type - node pool vs. custom - and the K8s distro to be able to reproduce it.

**Comment by HarrisonWAffel** [comment]
@catherineluse I was able to reproduce the issue only when creating a custom RKE2 cluster. I followed the following steps

1. Create custom RKE  2 cluster 
2. On the cluster registration screen click `Show Advanced`
![Screen Shot 2022-09-13 at 9 37 40 AM](https://user-images.githubusercontent.com/25062254/189916670-44e59585-031d-402e-a731-7181ad2b0a19.png)
3. Add in some taints and observe the registration command below update 
![Screen Shot 2022-09-13 at 9 37 51 AM](https://user-images.githubusercontent.com/25062254/189916831-794e3589-9a08-4c71-bed1-4e01f74615f2.png)
